tests/test_codecs.py

```diff
@@ -5,12 +5,38 @@
 import unittest
 from frontera.contrib.backends.remote.codecs.json import (Encoder as JsonEncoder,
                                                           Decoder as JsonDecoder,
-                                                          _convert, _reconvert)
+                                                          _encode_recursively, _decode_recursively)
 from frontera.contrib.backends.remote.codecs.msgpack import Encoder as MsgPackEncoder, Decoder as MsgPackDecoder
 from frontera.core.models import Request, Response
 import pytest
 
 
+def _compare_dicts(dict1, dict2):
+    """
+     Compares two dicts
+    :return: True if both dicts are equal else False
+    """
+    if dict1 == None or dict2 == None:
+        return False
+
+    if type(dict1) is not dict or type(dict2) is not dict:
+        return False
+
+    shared_keys = set(dict2.keys()) & set(dict2.keys())
+
+    if not (len(shared_keys) == len(dict1.keys()) and len(shared_keys) == len(dict2.keys())):
+        return False
+
+    dicts_are_equal = True
+    for key in dict1.keys():
+        if type(dict1[key]) is dict:
+            dicts_are_equal = dicts_are_equal and _compare_dicts(dict1[key], dict2[key])
```

Comment:
here you could return False earlier in case of inequality, there is no need to iterate till the end